File: 0066-plus-one/0066-plus-one.cpp
class Solution {
public:
    vector<int> plusOne(vector<int>& digits) {
        vector<int> ans;
        int carry = 1;
        int i = digits.size()-1;
        while(carry >0 || i >= 0){
            int sums;
            if(i >=0){
                sums = digits[i]+carry;
            }else{
                sums = carry;
            }
            carry = sums/10;
            ans.insert(ans.begin(),sums%10);
            i--;
        }
        return ans;
    }
};

File: 0066-plus-one/0066-plus-one.java
class Solution {
    public int[] plusOne(int[] digits) {
        int carry = 1;
        int n = digits.length -1;
        for(int i = n ; i>= 0 ; i--){
            int sum = carry + digits[i];
            digits[i] = sum%10;
            carry = sum/10;
        }
        if(carry==0) return digits;
        int[] ans = new int[digits.length+1];
        ans[0] = carry;
        System.arraycopy(digits, 0, ans, 1, digits.length);
        return ans;
    }
}